### ISSUE TITLE
Add support for escaping periods in block labels

### ISSUE THREAD

**Issue by mattdrees** [minamijoyo/hcledit#26]
I can't tell if there's a way to address a block that looks like this:
```hcl
provider "registry.terraform.io/hashicorp/aws" {
  version     = "3.38.0"
  constraints = "~> 3.38"
  hashes = [
    "h1:qKEjN/EM56XT46vGY33eoq7nD6JuGqRqFp7tkzTrRM0=",
  ]
}
```

The block can be listed, and the result looks like this:
```
$ cat .terraform.lock.hcl | hcledit block list
provider.registry.terraform.io/hashicorp/aws
```

But I can't use that (as is) to show the block:
```
$ cat .terraform.lock.hcl | hcledit block get provider.registry.terraform.io/hashicorp/aws

```

Probably the two dots are being parsed as part of the syntax for addresses. If I change those two to underscores, `block get` prints the block just fine:
```
$ cat .terraform.lock.hcl | hcledit block get provider.registry_terraform_io/hashicorp/aws
provider "registry_terraform_io/hashicorp/aws" {
  version     = "3.38.0"
  constraints = "~> 3.38"
  hashes = [
    "h1:qKEjN/EM56XT46vGY33eoq7nD6JuGqRqFp7tkzTrRM0=",
  ]
}
```

So clearly I need to escape those dots. But I can't figure out how. None of these work:
```
$ cat .terraform.lock.hcl | hcledit block get "provider.registry\.terraform\.io/hashicorp/aws"

$ cat .terraform.lock.hcl | hcledit block get "provider.registry\\.terraform\\.io/hashicorp/aws"

$ cat .terraform.lock.hcl | hcledit block get provider.'"registry.terraform.io/hashicorp/aws"'

$ cat .terraform.lock.hcl | hcledit block get provider."'registry.terraform.io/hashicorp/aws'"

```

Can you please add to the readme how to escape dots correctly?

**Comment by minamijoyo** [comment]
@mattdrees Thank you for opening this!

Unfortunately, there is no way to do this. The current implementation naively splits a given address by dots. The hcl specification doesn't define an address notation. I've chosen dots so that it's natural for Terraform users, but it had been before the Terraform added the dependency lock file.

According to the hcl spec, it's valid to use a dot in block labels, so we need to add support for escape dots in address.

**Comment by mattdrees** [comment]
Gotcha. Thank you.

I've updated the title to reflect that.

**Comment by beatcracker** [comment]
I've hacked up something to workaround this issue: https://github.com/beatcracker/hcledit/tree/address-delimiter

This patch adds support for a custom address delimiter, so you can do things like this:

```shell
hcledit --file .terraform.lock.hcl --delimiter ':' block get 'provider:registry.terraform.io/hashicorp/aws'
```

This is obviously a kludge and not a proper solution, but it gets the job done meanwhile.

**Comment by minamijoyo** [comment]
Fixed in v0.2.10 via #83. Thanks to @mcwarman 🎉
